lib/virt_autotest/esxi_utils.pm

```diff
@@ -73,13 +114,38 @@ sub disable_vm_time_synchronization {
-    console('svirt')->run_cmd("echo time.synchronize.continue=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
-    console('svirt')->run_cmd("echo time.synchronize.restore=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
-    console('svirt')->run_cmd("echo time.synchronize.resume.disk=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
-    console('svirt')->run_cmd("echo time.synchronize.shrink=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
-    console('svirt')->run_cmd("echo time.synchronize.tools.startup=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
-    console('svirt')->run_cmd("echo time.synchronize.resume.host=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
+    if (check_var('BACKEND', 'svirt')) {
+        console('svirt')->run_cmd("sed -ie 's/tools.syncTime.*/tools.syncTime=\"FALSE\"/' $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
+        console('svirt')->run_cmd("echo time.synchronize.continue=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
+        console('svirt')->run_cmd("echo time.synchronize.restore=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
+        console('svirt')->run_cmd("echo time.synchronize.resume.disk=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
+        console('svirt')->run_cmd("echo time.synchronize.shrink=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
+        console('svirt')->run_cmd("echo time.synchronize.tools.startup=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
+        console('svirt')->run_cmd("echo time.synchronize.resume.host=\"FALSE\" >> $vmx_file", domain => 'sshVMwareServer', wantarray => 1);
+    }
+    elsif (check_var('BACKEND', 'qemu')) {
+        script_run(qq(ssh -o StrictHostKeyChecking=no root\@$hypervisor "sed -ie 's/tools.syncTime.*/tools.syncTime=\"FALSE\"/' $vmx_file"));
+        script_run(qq(ssh -o StrictHostKeyChecking=no root\@$hypervisor "echo time.synchronize.continue=\"FALSE\" >> $vmx_file"));
+        script_run(qq(ssh -o StrictHostKeyChecking=no root\@$hypervisor "echo time.synchronize.restore=\"FALSE\" >> $vmx_file"));
+        script_run(qq(ssh -o StrictHostKeyChecking=no root\@$hypervisor "echo time.synchronize.resume.disk=\"FALSE\" >> $vmx_file"));
+        script_run(qq(ssh -o StrictHostKeyChecking=no root\@$hypervisor "echo time.synchronize.shrink=\"FALSE\" >> $vmx_file"));
+        script_run(qq(ssh -o StrictHostKeyChecking=no root\@$hypervisor "echo time.synchronize.tools.startup=\"FALSE\" >> $vmx_file"));
+        script_run(qq(ssh -o StrictHostKeyChecking=no root\@$hypervisor "echo time.synchronize.resume.host=\"FALSE\" >> $vmx_file"));
+    }
+}
+
+sub recover_vmx_config {
```

Comment:
Seems this function only reset time related config, not all config? How about making the function name more specific to reflect this?

Reply:
How about the name 'revert_vm_timesync_setting'?